src/Netscript/NetscriptHelpers.tsx

```diff
@@ -716,3 +724,32 @@ let customElementKey = 0;
 export function wrapUserNode(value: unknown) {
   return <CustomBoundary key={`PlayerContent${customElementKey++}`}>{value}</CustomBoundary>;
 }
+
+function validateBitNodeOptions(ctx: NetscriptContext, bitNodeOptions: unknown): BitNodeOptions {
+  const result = getDefaultBitNodeOptions();
+  if (bitNodeOptions === undefined || bitNodeOptions === null) {
+    return result;
+  }
+  if (typeof bitNodeOptions !== "object") {
+    throw errorMessage(ctx, `bitNodeOptions must be an object if it's specified. It was ${bitNodeOptions}.`);
+  }
+  const options = bitNodeOptions as Unknownify<BitNodeOptions>;
+  if (!(options.activeSourceFiles instanceof Map)) {
+    throw errorMessage(ctx, `activeSourceFiles must be a Map.`);
+  }
+  const validationResultForActiveSourceFiles = validateActiveSourceFiles(options.activeSourceFiles);
+  if (!validationResultForActiveSourceFiles.valid) {
+    throw errorMessage(ctx, `activeSourceFiles is invalid. Reason: ${validationResultForActiveSourceFiles.message}`);
+  }
+
+  result.activeSourceFiles = new Map(options.activeSourceFiles);
```

Comment:
Feels like this should be a JSONMap? That's strictly more useful, and no more expensive.